pkg/gpu/safenvml/device.go

```diff
@@ -83,6 +83,10 @@ type SafeDevice interface {
 	GetUtilizationRates() (nvml.Utilization, error)
 	// IsMigDeviceHandle returns true if the device is a MIG device or false for a physical device
 	IsMigDeviceHandle() (bool, error)
+	// GpmQueryDeviceSupport returns true if the device supports GPM
+	GpmQueryDeviceSupport() (nvml.GpmSupport, error)
+	// GpmSampleGet gets a sample for GPM
+	GpmSampleGet(sample nvml.GpmSample) error
```

Comment:
I think you should also add `GpmMigSampleGet` ([ref](https://docs.nvidia.com/deploy/nvml-api/group__nvmlGpmFunctions.html#group__nvmlGpmFunctions_1ge7880c391d245f640e5cb7633c6db1fe)) to support MIG devices